isocontour improvements

chrishavlin opened the pull request:
The main change in this PR is the ability to set the alpha channel for isocontours individually from the user interface. 

Also includes:
* general fixes for isocontours (they weren't actually rendering for me anymore)
* unrelated typo fix in `opengl_support`

Draft for now as I'd like to:
* add some minimal tests for the isocontours
* re-consider how we're scaling/selecting isocontour values and dealing with user specified log values.

matthewturk commented on yt_idv/opengl_support.py:
```diff
@@ -440,7 +440,7 @@ def _viewport_default(self):
     @traitlets.observe("viewport")
     def _viewport_changed(self, change):
         # we just need to disable the initialized value here
-        self.initalized = False
```
well this seems like a big deal

matthewturk commented on yt_idv/scene_components/base_component.py:
```diff
@@ -30,6 +30,7 @@ class SceneComponent(traitlets.HasTraits):
     use_db = traitlets.Bool(False)  # use depth buffer
     iso_tolerance = traitlets.CFloat(0.025)
     iso_layers = traitlets.List()
+    iso_layers_alpha = traitlets.List()
```
This doesn't need to change for this, but I'm genuinely wondering if we should make `iso_layers` a `List` of `Tuple`s.

chrishavlin replied:
ya, I was kind of thinking that as well! Might make it more flexible for changes down the line too. e.g., adding a "visible" toggle for each layer.

chrishavlin replied:
So I decided not to make this change here, see my note about a wider re-factor in the latest comment.

matthewturk reviewed (approved):
lgtm

chrishavlin commented:
I think this is at a good stage for final review + merge. 

I ended up doing a wider refactor beyond adding the alpha channel. To summarize the changes :

* I moved the isocontour settings to their own UI tree that appears when selecting the isocontour shader
* I added a separate log-toggle for the isocontours to specify whether you want to be setting base 10 exponents or actual values for isocontour values. This is now separate from the cmap log-toggle (so you can set exponents but have the cmap visualize in log-scale).
* I changed the isocontour tolerance behavior to allow for two ways of setting the tolerance:
	1. absolute value range, i.e., identify values that are within target value +/- acceptible range in terms of actual data values, not normalized data values. This is good for ~linear data.
	2. a percent deviation from a target value. i.e., select values within x% of target value. This means that the tolerance in a raw data range will actually be different for each layer, which makes it much easier to add multiple layers for log-varying data. 
	
Here's an example of selecting three layers, 10^-29, 10^-30 and 10^-28 with tolerance within +/-1% of the target values:
	

![isocontours](https://user-images.githubusercontent.com/22038879/146578893-fd4e6650-9f07-4880-ae29-813a79cf901e.png)


Note, I did not end up adding tests or moving the isocontour attributes into a single tuple.... because I think a larger refactor is warranted: I think we should move the isocontour behavior to its own scene_component or scene_annotation class. I'm not sure on which, I think its behavior is kind of in between the two: it's data behavior is more like the scene_components as it still needs the full data textures to render, but the user interaction feels more like a scene_annotation. And it'd be nice to be able to use one of the volume rendering shaders while also rendering isocontours... in any case, that feels much further beyond the scope of this PR.
